<launch>

  <arg name="gui" default="true" />

  <node name="image_publisher"
        pkg="jsk_perception" type="image_publisher.py">
    <remap from="~output" to="image" />
    <param name="file_name" value="$(find jsk_perception)/sample/ros_fuerte.jpg" />
  </node>

  <node name="slic_super_pixels"
        pkg="jsk_perception" type="slic_super_pixels" >
    <param name="publish_debug_images" value="$(arg gui)" />
  </node>

  <group if="$(arg gui)">
    <node name="image_view0"
          pkg="image_view" type="image_view">
      <remap from="image" to="slic_super_pixels/debug" />
    </node>
    <node name="image_view1"
          pkg="image_view" type="image_view">
      <remap from="image" to="slic_super_pixels/debug/mean_color" />
    </node>
    <node name="image_view2"
          pkg="image_view" type="image_view">
      <remap from="image" to="slic_super_pixels/debug/center_grid" />
    </node>
  </group>

</launch>
